02-peer-protocol.md

```diff
@@ -566,14 +566,17 @@ still be under the maximum message size.  It also ensures that
 a single penalty transaction can spend the entire commitment transaction,
 as calculated in [BOLT #5](05-onchain.md#penalty-transaction-weight-calculation).
 
-### Removing an HTLC: `update_fulfill_htlc` and `update_fail_htlc`
+### Removing an HTLC: `update_fulfill_htlc`, `update_fail_htlc` and `update_fail_malformed_htlc`
 
 For simplicity, a node can only remove HTLCs added by the other node.
 There are three reasons for removing an HTLC: it has timed out, it has
 failed to route, or the payment preimage is supplied.
 
 The `reason` field is an opaque encrypted blob for the benefit of the
-original HTLC initiator as defined in [BOLT #4](04-onion-routing.md).
+original HTLC initiator as defined in [BOLT #4](04-onion-routing.md),
+but there's a special malformed failure variant for the case where
+our peer couldn't parse it; in this case the current node encrypts
```

Comment:
Let's be clear,  C will see a bad MAC, and send ipdate_fail_malformed_htlc with  BADONION|PERM|6 + sha256(onion).  B could (in theory) check this sha, see C is on crack, and re-send (or close all channels with C because it's wigging out, and send back a permanent failure).  But if it doesn't, it encapsulates this into a normal fail message and pushed it back to A.

Now, A doesn't know if it's B or C's fault that it got garbage (it sees the SHA is wrong), but it doesn't really care: it discards that B->C channel from consideration and retries.  Which is kind of the Right Thing.